fix: issues with menu component

apvale opened the pull request:
## Description of the change

Fixes issues with the Menu component and adds minW and maxW as shown in Figma styles. 

Before: 

<img width="787" alt="image" src="https://github.com/Localitos/pluto/assets/5320963/ce97b25f-3d97-4c27-a969-a49d4197dbe7">

After:

<img width="988" alt="image" src="https://github.com/Localitos/pluto/assets/5320963/859fcfef-da03-4604-95df-363df092ad8e">


## Testing the change

- [ ] Write your testing instructions here

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Non-Breaking Change (change to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Development

- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached.

nyan07 commented on .changeset/chilly-geese-reflect.md:
```diff
@@ -0,0 +1,8 @@
+[Menu] Fix issues on Menu component
+
+- Remove extra padding
+- Use outline instead of border
```
I would describe what it fixes, not what the solution was.

rssilva commented on packages/components/src/components/Menu/Menu.tsx:
```diff
@@ -50,14 +50,15 @@ const Menu = React.forwardRef<HTMLButtonElement, MenuProps>(
             flexDirection="column"
-            paddingBottom="space20"
-            paddingTop="space20"
+            maxWidth="280px"
```
Late to the party here but shouldn't this maxWidth be configurable as a prop? If we have text a bit bigger as option will the line break?

apvale replied:
Thank you for the heads up, Rafa. 🙏🏻 
I spoke with Hanna, who said the option should break the line when reaching the maxWidth. 

<img width="428" alt="image" src="https://github.com/Localitos/pluto/assets/5320963/ea4ec781-c84c-4e4f-95c5-14a866de8151">

apvale commented:
@nyan07 @rssilva Can you check again the PR after I changed a bit the Menu component? https://github.com/Localitos/pluto/pull/1108/commits/30f0f1435a6bac572bee0a6794772ee6b867313d
I hope the changeset is improved. But I noticed that the item wasn't triggering the onClick, only if I clicked exactly in the button. So, I moved the trigger to the whole menu item.
